Add evaluation script

ericagreene opened the pull request:
The goal here is to be able to train a model, save it and then later load the model and run new data through it. This sequence involves saving the `VocabularyProcessor` to disk and pre-processing the new data using it. 

WikiData now takes a `predict_mode` flag. If `predict_mode=True`, then we process the data using a pre-trained `VocabularyProcessor`. 

**To Test**
```
./bin/run_local
python trainer/eval.py --model_dir=[path-to-saved-model] --test_data=local_data/test.csv
```

Right now, `eval.py` just prints out the predictions for the first 10 examples. This is not very useful for now, but we can make it more useful in the future.

nthain commented on kaggle-classification/trainer/eval.py:
```diff
@@ -0,0 +1,93 @@
+    """
+    model_inputs = []
+
+    for d in data:
```
Let's change this to use the test_input_fn and classifier.predict(...) approach that we use in the model.py file.

ericagreene replied:
I looked into this, and it turns out that the `export_savedmodel` method on `Estimator` exports a `tensorflow.contrib.predictor.saved_model_predictor.SavedModelPredictor`, not an `Estimator`. Docs [here](https://www.tensorflow.org/api_docs/python/tf/contrib/learn/Estimator#export_savedmodel). So we can't use the same `classifier.predict()` approach to run data through the model. I couldn't figure out how to save an `Estimator` as an `Estimator`, but would love a pointer if you know how to do that because I agree it would be easier!

ericagreene commented on kaggle-classification/trainer/eval.py:
```diff
@@ -0,0 +1,93 @@
+def predict(data, classifier):
+    """
+    Args:
+      data: n x m array of features where n is the number of examples
```
Not multiplication, not sure what the correct way to write this is. I want to say "a n times m" array. 
`n x m`?
`n * m`?

iislucas replied:
I think `'n * m' shaped array...` is probably clearest.

ericagreene commented:
Talked offline and decided to add try/catch logic when loading the model in case we're accidentally trying to load a Keras model, which isn't supported yet. And make a separate issue to add functionality to load a Keras model.

ericagreene commented:
@nthain I just rebased this branch and updated the documentation. 

To test, `bin/run_local` to train a new model. The run `python trainer/eval.py --model_dir=[path-to-saved-model] --test_data=local_data/test.csv` to run the eval. It should evaluate the first 100 examples in the test dataset and write the results to the `model-dir` directory.

ericagreene commented:
Closing this since it's out of date and we have much better possible solutions now.
